oxd-server/src/main/java/org/gluu/oxd/server/service/TokenService.java

```diff
@@ -0,0 +1,188 @@
+package org.gluu.oxd.server.service;
+
+import com.google.common.collect.Lists;
+import com.google.inject.Inject;
+import org.gluu.oxauth.client.*;
+import org.gluu.oxauth.model.common.AuthenticationMethod;
+import org.gluu.oxauth.model.common.GrantType;
+import org.gluu.oxauth.model.common.Prompt;
+import org.gluu.oxauth.model.common.ResponseType;
+import org.gluu.oxauth.model.uma.UmaScopeType;
+import org.gluu.oxauth.model.util.Util;
+import org.gluu.oxd.common.CoreUtils;
+import org.gluu.oxd.common.ErrorResponseCode;
+import org.gluu.oxd.server.HttpException;
+import org.gluu.oxd.server.OxdServerConfiguration;
+import org.gluu.oxd.server.model.Pat;
+import org.gluu.oxd.server.model.Token;
+import org.gluu.oxd.server.model.TokenFactory;
+import org.gluu.oxd.server.model.UmaTokenFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+public class TokenService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TokenService.class);
+
+    private final RpService rpService;
+    private final ValidationService validationService;
+    private final DiscoveryService discoveryService;
+    private final HttpService httpService;
+    private final OxdServerConfiguration configuration;
+    private final StateService stateService;
+
+    @Inject
+    public TokenService(RpService rpService,
```

Comment:
This whole class is just copy paste of UmaTokenService. I vote to generalize existing code inside UmaTokenService and then just rename it to TokenService. We don't need both with copy/pastes. Like:
```
    private Token obtainTokenWithClientCredentials(OpenIdConfigurationResponse discovery, Rp site, UmaScopyType umaScopeType) {
  if (umaScopeType == null) {
        // tune scopes here
  }
  // perform check depending on requested scopes
```